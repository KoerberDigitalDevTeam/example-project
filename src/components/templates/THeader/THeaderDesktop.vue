<template>
  <div class="theader-desktop__container">
    <h3 v-testId.theader-desktop.title class="theader-desktop__title">{{ title }}</h3>
  </div>
</template>

<script>
export default {
  name: 'THeaderDesktop',

  props: {
    title: {
      type: String,
      required: true,
    },
  },
};
</script>

<style lang="scss" scoped>
.theader-desktop__container {
  padding: 16px 0 16px 24px;
}
</style>
