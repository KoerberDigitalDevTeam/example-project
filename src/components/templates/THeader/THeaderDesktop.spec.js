import { shallowMount } from '@vue/test-utils';
import Vuetify from 'vuetify';
import THeaderDesktop from './THeaderDesktop.vue';

function createWrapper(options = {}) {
  return shallowMount(THeaderDesktop, {
    vuetify: new Vuetify(),
    ...options,
  });
}

describe('THeaderDesktop', () => {
  it('Should render THeaderDesktop', () => {
    const wrapper = createWrapper({
      propsData: {
        title: 'Title',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe('Template:', () => {
    it('Should render the `title`', () => {
      const title = 'Random Title';
      const wrapper = createWrapper({
        propsData: {
          title,
        },
      });
      const titleElement = wrapper.find(`[data-testid=theader-desktop-title]`);
      expect(titleElement.text()).toBe(title);
    });
  });

  describe('Props Interface:', () => {
    it('Should comply with the `title` prop interface', () => {
      expect(THeaderDesktop.props.title).toMatchObject({
        type: String,
        required: true,
      });
    });
  });
});
