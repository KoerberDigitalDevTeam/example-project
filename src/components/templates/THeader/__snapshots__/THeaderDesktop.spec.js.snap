// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`THeaderDesktop Should render THeaderDesktop 1`] = `
<div class="theader-desktop__container">
  <h3 class="theader-desktop__title" data-testid="theader-desktop-title">Title</h3>
</div>
`;
